code/modules/modular_computers/file_system/programs/engineering/sm_monitor.dm

```diff
@@ -0,0 +1,137 @@
+/datum/computer_file/program/supermatter_monitor
+	filename = "smmonitor"
+	filedesc = "Supermatter Monitoring"
+	ui_header = "smmon_0.gif"
+	program_icon_state = "smmon_0"
+	extended_desc = "This program connects to specially calibrated supermatter sensors to provide information on the status of supermatter-based engines."
+	requires_ntnet = TRUE
+	transfer_access = access_engine
+	network_destination = "supermatter monitoring system"
+	size = 5
+	requires_ntnet = 1
+	var/last_status = SUPERMATTER_INACTIVE
+	var/list/supermatters
+	var/obj/machinery/power/supermatter_shard/active		// Currently selected supermatter crystal.
+
+
+/datum/computer_file/program/supermatter_monitor/process_tick()
+	..()
+	var/new_status = get_status()
+	if(last_status != new_status)
+		last_status = new_status
+		ui_header = "smmon_[last_status].gif"
+		program_icon_state = "smmon_[last_status]"
+		update_computer_icon()
+
+/datum/computer_file/program/supermatter_monitor/run_program(mob/living/user)
+	. = ..(user)
+	refresh()
+
+/datum/computer_file/program/supermatter_monitor/kill_program(forced = FALSE)
+	active = null
+	supermatters = null
+	..()
+
+// Refreshes list of active supermatter crystals
+/datum/computer_file/program/supermatter_monitor/proc/refresh()
+	supermatters = list()
+	var/turf/T = get_turf(nano_host())
+	if(!T)
+		return
+	for(var/obj/machinery/power/supermatter_shard/S in atmos_machinery)
+		// Delaminating, not within coverage, not on a tile.
+		if(!(S.z == 1 || S.z == 5 || S.z == T.z) || !istype(S.loc, /turf/simulated/))
```

Comment:
The first two need to use helper proces for checking the station or mining z-levels explicitly.